### PR TITLE
FEEval: replace usages of n_vectorization_actual

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

File: include/deal.II/matrix_free/fe_evaluation.h [hunk 1]
```diff
@@ -3625,8 +3647,12 @@ FEEvaluationBase<dim, n_components_, Number, is_face, VectorizedArrayType>::
     for (unsigned int comp = 0; comp < n_components; ++comp)
       for (unsigned int i = 0; i < dofs_per_component; ++i)
         operation.process_empty(values_dofs[comp][i]);
```
Comment (kronbichler) [inline include/deal.II/matrix_free/fe_evaluation.h:3649]: If I see things correctly, this is the only place where we still use `n_vectorization_actual` (line 3646). Would it make sense to move the definition of line 3471 here?

File: include/deal.II/matrix_free/fe_evaluation.h [hunk 2]
```diff
@@ -6874,8 +6901,12 @@ FEEvaluation<dim,
   this->jacobian = &this->mapping_data->jacobians[0][offsets];
   this->J_value  = &this->mapping_data->JxW_values[offsets];
 
-  for (unsigned int i = 0; i < VectorizedArrayType::size(); ++i)
+  unsigned int i = 0;
+  for (; i < this->matrix_free->n_active_entries_per_cell_batch(this->cell);
```
Comment (kronbichler) [inline include/deal.II/matrix_free/fe_evaluation.h:6905]: I think we might need to see over https://github.com/dealii/dealii/blob/668e626b5edf58149cfb5e6fd74ece09a9a8b6df/include/deal.II/matrix_free/matrix_free.h#L2581-L2589 which implements the `n_active_entries_...` function, in order to limit costs to a single index lookup in these performance-critical places. (For example, `dof_info` stores this number, but we allow to not set it up.)

File: include/deal.II/matrix_free/fe_evaluation.h [hunk 3]
```diff
@@ -7596,8 +7627,13 @@ FEFaceEvaluation<dim,
              "is_interior_face set to true. "));
 
   this->reinit_face(this->matrix_free->get_face_info(face_index));
-  for (unsigned int i = 0; i < VectorizedArrayType::size(); ++i)
+
+  unsigned int i = 0;
+  for (; i < this->matrix_free->n_active_entries_per_face_batch(this->cell);
```
Comment (kronbichler) [inline include/deal.II/matrix_free/fe_evaluation.h:7632]: Same here, we need to see the implementation over at some point.

File: include/deal.II/matrix_free/fe_evaluation.h (hunk 1 above)
Comment (peterrum) [reply include/deal.II/matrix_free/fe_evaluation.h:3649]: I have moved it down.

File: include/deal.II/matrix_free/fe_evaluation.h (hunk 2 above)
Comment (peterrum) [reply include/deal.II/matrix_free/fe_evaluation.h:6905]: I have created an issue #13263.

File: include/deal.II/matrix_free/fe_evaluation.h [hunk 4]
```diff
@@ -3461,19 +3464,21 @@ FEEvaluationBase<dim, n_components_, Number, is_face, VectorizedArrayType>::
...
-    this->dof_info
-      ->n_vectorization_lanes_filled[this->dof_access_index][this->cell];
-
   const std::array<unsigned int, VectorizedArrayType::size()> &cells =
     this->get_cell_ids();
 
+  const unsigned int n_vectorization_actual =
+    std::count_if(cells.begin(), cells.end(), [](const auto i) {
+      return i != numbers::invalid_unsigned_int;
+    });
```
Comment (kronbichler) [inline include/deal.II/matrix_free/fe_evaluation.h:3473]: You still have this code in - did you push an update yet?

Comment (peterrum) [reply include/deal.II/matrix_free/fe_evaluation.h]: Now!

File: include/deal.II/matrix_free/fe_evaluation.h (hunk 1 above)
Comment (kronbichler) [reply include/deal.II/matrix_free/fe_evaluation.h:3649]: I have one more request: I see that this loop here in lines 3646-3649 is exactly the same as the one in line 3604-3607. Can you move the former one out of the `if`? In that case, I would then further prefer to pull the computation into the `if, i.e., something like `if(std::count_if(...) < n_lanes)` (unless it looks very ugly).

Comment (peterrum) [reply include/deal.II/matrix_free/fe_evaluation.h]: Done! I though normally you don't like the lambdas in if-statements ;)

**Review by kronbichler — APPROVED** [review]

**Pull request merged**